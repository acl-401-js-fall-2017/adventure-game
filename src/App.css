.App {
  text-align: center;
}

.App-logo {
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 30px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}