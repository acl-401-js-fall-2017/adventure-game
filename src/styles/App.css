.App {
  text-align: center; }

.App-logo {
  height: 80px;
  width: auto;
  border: 2px solid black; }

.App-header {
  background-color: #d2691e;
  height: 140px;
  padding: 5px 35px 40px 35px;
  color: black; }
  .App-header > h2 {
    font-style: italic; }

.App-title {
  font-family: New Century Schoolbook, serif;
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

.challenge .mission {
  font-family: serif;
  font-style: italic;
  font-size: 2em; }

.challenge .message {
  font-family: sans-serif;
  font-weight: bold; }

#nextButton {
  font-weight: bolder;
  background-color: green;
  border-radius: 15px; }
  #nextButton:hover {
    background-color: greenyellow; }

#prevButton {
  font-weight: bolder;
  background-color: #dd9c0f;
  border-radius: 15px; }
  #prevButton:hover {
    background-color: #e7030f; }
