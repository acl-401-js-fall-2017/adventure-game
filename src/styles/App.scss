.App {
  text-align: center;
}

.App-logo {
  height: 80px;
  width: auto;
  border: 2px solid black;
}

.App-header {
  background-color: #d2691e;
  height: 140px;
  padding: 5px 35px 40px 35px;
  color: black;
  > h2 {
    font-style: italic;
  }
}


.App-title {
  font-family: New Century Schoolbook,
  serif;
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.challenge{
  .mission {
    font-family: serif;
    font-style: italic;
    font-size: 2em;
  }
  .message {
    font-family: sans-serif;
    font-weight: bold;
  }
}

#nextButton {
  font-weight: bolder;
  background-color: green;
  border-radius: 15px;
  &:hover {
    background-color: greenyellow;
  }
}

#prevButton {
  font-weight: bolder;
  background-color: rgb(221, 156, 15);
  border-radius: 15px;
  &:hover {
    background-color: rgb(231, 3, 15);
  }
}

