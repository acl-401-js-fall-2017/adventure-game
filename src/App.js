import React, { Component } from 'react';
import logo from './pics/goldenTx.jpg';
import Candyshop from './components/Candyshop';
import Gate from './components/Gate';
import Chocolate from './components/Chocolate';
import Inventing from './components/Inventing';
import Staff from './components/Staff';

import './App.css';

class App extends Component {
  
  constructor() {
    super();
    this.state = {
      inventory: [],
      room: 0
    };
  }

  handleInventory = item => {
    const inventory = this.state.room ===3 ? [] : [ ...this.state.inventory ];
    inventory.push(item);
    this.setState({
      inventory
    });
  }
  
  removeInventory = () => {
    const inventory = [...this.state.inventory];
    inventory.splice(-1);
    this.setState({
      inventory
    });
  }

  changeRoom = ({ target }) => {
    this.setState(prevState => {
      let room;
      if (target.id === 'prevButton') {
        room = prevState.room - 1;
      }
      if (target.id === 'nextButton') {
        room = prevState.room + 1;
      }
      return { room };
    });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">The Wonka Tour</h1>
          <h2> Your Inventory: {this.state.inventory} </h2>
        </header>
        {this.state.room === 0 && (
          <Candyshop handleInventory = {this.handleInventory} inventory = {this.state.inventory} />
        )}
        {this.state.room === 1 && (
          <Gate removeInventory={this.removeInventory} inventory={this.state.inventory}/>
        )}
        {this.state.room === 2 && (
          <Chocolate handleInventory={this.handleInventory} inventory={this.state.inventory} />
        )}
        {this.state.room === 3 && (
          <Inventing handleInventory={this.handleInventory} inventory={this.state.inventory} />
        )}
        {this.state.room === 4 && (
          <Staff  removeInventory={this.removeInventory} inventory={this.state.inventory} />
        )}
        {this.state.room !== 0 && (
          <button id="prevButton" onClick={this.changeRoom} >Previous room </button>
        )}
        {this.state.room !== 4 && (
          <button id="nextButton" onClick={this.changeRoom} >Next room </button>
        )}
      </div>
    );
  }
}

export default App;