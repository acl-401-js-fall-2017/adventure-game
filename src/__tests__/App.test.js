import React from 'react';

import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import App from '../App';

describe('Game', () => {

  it('shows a app', () => {
    const wrapper = shallow(<App/>);
    console.log(toJSON(wrapper));
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});