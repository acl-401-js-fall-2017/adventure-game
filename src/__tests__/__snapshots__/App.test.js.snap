// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game shows a app 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="goldenTx.jpg"
    />
    <h1
      className="App-title"
    >
      The Wonka Tour
    </h1>
    <h2>
       Your Inventory: 
       
    </h2>
  </header>
  <Candyshop
    handleInventory={[Function]}
    inventory={Array []}
  />
  <button
    id="nextButton"
    onClick={[Function]}
  >
    Next room 
  </button>
</div>
`;
